src/main/java/by/custom/utilcalculator/service/UserProgressManager.java

```diff
@@ -26,31 +26,99 @@ public static UserProgressManager getInstance() {
+            //self-propelled vehicles
+            case GRADERS -> userProgress.setParticularTransportType(ParticularTransportType.GRADER);
+            case BULLDOZERS -> userProgress.setParticularTransportType(ParticularTransportType.BULLDOZER);
+            case EXCAVATORS -> userProgress.setParticularTransportType(ParticularTransportType.EXCAVATOR);
+            case WHEEL_LOADERS -> userProgress.setParticularTransportType(ParticularTransportType.WHEEL_LOADER);
+            case TAMPING_MACHINES -> userProgress.setParticularTransportType(ParticularTransportType.TAMPING_MACHINE);
+            case FRONT_LOADERS -> userProgress.setParticularTransportType(ParticularTransportType.FRONT_LOADER);
+            case WHEELED_CRANES -> userProgress.setParticularTransportType(ParticularTransportType.WHEELED_CRANES);
+            case PIPELAYERS -> userProgress.setParticularTransportType(ParticularTransportType.PIPELAYERS);
+            case TRAILERS_OTHER -> userProgress.setParticularTransportType(ParticularTransportType.TRAILERS_OTHER);
+            case ROAD_MAINTENANCE -> userProgress.setParticularTransportType(ParticularTransportType.ROAD_MAINTENANCE);
+            case FORESTRY -> userProgress.setParticularTransportType(ParticularTransportType.FORESTRY);
+            case FORWADERS -> userProgress.setParticularTransportType(ParticularTransportType.FORWADERS);
+            case TIMBER_LOADERS -> userProgress.setParticularTransportType(ParticularTransportType.TIMBER_LOADERS);
+            case WHEELED_TRACTORS -> userProgress.setParticularTransportType(ParticularTransportType.WHEELED_TRACTORS);
+            case CRAWLER_TRACTORS -> userProgress.setParticularTransportType(ParticularTransportType.CRAWLER_TRACTORS);
+            case COMBINE_HARVESTERS ->
+                    userProgress.setParticularTransportType(ParticularTransportType.COMBINE_HARVESTERS);
+            case FORAGE_HARVESTERS ->
+                    userProgress.setParticularTransportType(ParticularTransportType.FORAGE_HARVESTERS);
+            case AGRICULTURAL_VEHICLES ->
+                    userProgress.setParticularTransportType(ParticularTransportType.AGRICULTURAL_VEHICLES);
+            case OFF_ROAD_DUMP_TRUCKS ->
+                    userProgress.setParticularTransportType(ParticularTransportType.OFF_ROAD_DUMP_TRUCKS);
+        }
+
+        userProgressStorage.save(userProgress);
+        return messagesCreator.getSummaryAnswer(userProgress);
+    }
+
+    public String processN1_N3TransportWeight(final Command requestingCommand, final String chatID) throws UtilsborException {
+        final UserProgress userProgress = userProgressStorage.get(chatID);
+
+        if (!CommandTree.validateCommand(requestingCommand, userProgress)) {
+            throw new InvalidOrderCommandException(chatID, requestingCommand);
+        }
+
+        switch (requestingCommand) {
+            case LESS_2_TONS -> userProgress.setWeight(Weight.LESS_2_TONS);
```

Comment:
You already have association between command and StepIndecator. I think you can use instead duplicate this in switch-case

Reply:
🤔 it looks very promising, I'll try it